yarn-project/noir-contracts/src/contracts/ecdsa_account_contract/src/main.nr

```diff
@@ -61,9 +74,24 @@ contract EcdsaAccount {
         for byte in signing_pub_key_y { args = args.push(byte as Field); }
         let mut context = Context::new(inputs, abi::hash_args(args.storage));
         
-        let pub_key_note = EcdsaPublicKeyNote::new(signing_pub_key_x, signing_pub_key_y, inputs.call_context.storage_contract_address);
+        let this = inputs.call_context.storage_contract_address;
+        let pub_key_note = EcdsaPublicKeyNote::new(signing_pub_key_x, signing_pub_key_y, this);
         context = storage.public_key.initialise(context, pub_key_note);
         
+        context = emit_encrypted_log(
+            context,
+            this,
+            storage.public_key.storage_slot,
+            get_public_key(this),
+            pub_key_note.serialise(),
+        );
```

Comment:
@LeilaWang should this be part of `ImmutableSingleton#initialise`? Or is there a scenario where we may want to initialise it without emitting the corresponding encrypted log?

Reply:
I actually thought about the same - emitting a log when a note is created (in singleton.initialise, set.insert, etc). But then decided to keep it separate. Because in some use cases we might want to emit unencrypted logs instead of encrypted ones. And in some use cases, like you said, we might not want to emit anything at all (for example, the ClaimNote in zk_token_contract).

Reply:
I've added a header to a note in a recent PR, and will change the emit apis to be cleaner since we won't be passing context, contract address and storage slot around :)